api/src/main/java/org/eclipse/microprofile/lra/annotation/Complete.java

```diff
@@ -36,4 +37,23 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.METHOD})
 public @interface Complete {
+    /**
+     * The period for which the participant will guarantee it will be able
+     * to complete for any work that it performed during the associated LRA.
+     * When this period elapses the LRA that it joined becomes eligible for
+     * cancellation. The units are specified in the {@link Complete#timeUnit()}
+     * attribute.
+     *
+     * A value of zero indicates that it will always be able to complete.
+     *
+     * @return the period for which the participant can guarantee it
+     * will be able to complete when asked to do so
+     */
+    long timeLimit() default 0;
+
+    /**
+     * @return the unit of time that the {@link Complete#timeLimit()} attribute is
+     * measured in.
+     */
+    TimeUnit timeUnit() default TimeUnit.SECONDS;
```

Comment:
there is nothing wrong with adding this capability to Complete also but my concern is that LRAClient join method will now need two timelimit parameters to stay aligned with functionality available by the annotations - https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/client/LRAClient.java#L278

Reply:
Use ChronoUnit instead of TimUnit to be consistent with other MP specs.

Reply:
We've decided on the mtg that LRAClient#joinLRA taking individual URLs and timelimit will be removed, so my previous comment is invalid